packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -162,8 +162,40 @@ function extractImportedBindings(
  */
 export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
   const { root, base } = config
+  const needCheckFileExistsRE = new RegExp(root + '/(?!node_modules/)')
   const clientPublicPath = path.posix.join(base, CLIENT_PUBLIC_PATH)
   const enablePartialAccept = config.experimental?.hmrPartialAccept
+  const filepathExistsCache = new Map()
```

Comment:
I wonder if we're concerned about this growing too large in size? If we're going to use it then it might be nice to share other places we're checking if a path exists? And so maybe it should live someplace more common like the `utils` class I'm referring to below so it can be used more widely if we do want to keep it